server/src/controllers/products/getProductByID.ts

```diff
@@ -0,0 +1,20 @@
+import { NextFunction, Request, Response } from 'express';
+import { Product, Category } from '../../database';
+
+const getProductByID = async (req:Request, res:Response, next:NextFunction) => {
+  const { id } = req.params;
+  try {
+    const productByID = await Product.findOne({ include: Category, where: { id } });
+    if (productByID) {
+      res.json({ success: true, data: productByID });
+    } else {
+      res.status(404).json({ success: false, message: 'No product found!' });
+    }
+  } catch (err: any) {
+    if (err.details) {
```

Comment:
ensure that you don't send twice.
```suggestion
if (err.details) {
    return  res.status(422).json({ msg: err.details[0].message, status: 422 });
    }
  return  next(err);
  }```